core/src/components/content/test/custom/index.html

```diff
@@ -0,0 +1,201 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+
+<head>
+  <meta charset="UTF-8">
+  <title>Content - Custom</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
+  <link href="../../../../../css/ionic.bundle.css" rel="stylesheet">
+  <link href="../../../../../scripts/testing/styles.css" rel="stylesheet">
+  <script src="../../../../../scripts/testing/scripts.js"></script>
+  <script nomodule src="../../../../../dist/ionic/ionic.js"></script>
+  <script type="module" src="../../../../../dist/ionic/ionic.esm.js"></script>
+</head>
+
+<body>
+  <ion-app mode="ios">
+    <ion-header collapse="fade">
+      <ion-toolbar>
+        <ion-buttons slot="start">
+          <ion-button onclick="scrollToTop()">
+            <ion-icon slot="icon-only" name="arrow-up"></ion-icon>
+          </ion-button>
+        </ion-buttons>
+
+        <ion-title onclick="statusTap()">Content - Custom</ion-title>
+
+        <ion-buttons slot="end">
+          <ion-button onclick="scrollToBottom()">
+            <ion-icon slot="icon-only" name="arrow-down"></ion-icon>
+          </ion-button>
+        </ion-buttons>
+      </ion-toolbar>
+    </ion-header>
+
+    <div ion-content style="height: 100%; position: relative; overflow: hidden;" id="content">
+      <div id="background-content"></div>
+      <div style="position: absolute; left: 0; top: 0; width: 100%;">
+        <ion-refresher slot="fixed" id="ion-refresher">
+          <ion-refresher-content></ion-refresher-content>
+        </ion-refresher>
+      </div>
+      <div id="inner-scroll">
+        <f style="height: 50%"></f>
+        <f></f>
+        <f></f>
+        <f></f>
+        <f></f>
+        <f></f>
+        <f></f>
+        <f></f>
+        <ion-list>
+          <ion-item>
+            <ion-label>Input</ion-label>
+            <ion-input></ion-input>
+          </ion-item>
+        </ion-list>
+
+        <f></f>
+
+        <ion-infinite-scroll>
+        </ion-infinite-scroll>
+      </div>
+    </div>
+
+    <ion-footer collapse="fade">
+      <ion-toolbar>
+        <ion-title>Footer</ion-title>
+
+        <ion-buttons slot="end">
+          <ion-button>
+            <ion-icon slot="icon-only" name="help-circle-outline"></ion-icon>
+          </ion-button>
+        </ion-buttons>
+      </ion-toolbar>
+    </ion-footer>
+
+    <script>
+      const content = document.getElementById('content');
+
+      content.getScrollElement = async function() {
```

Comment:
I need to put a little extra thought into the API design around this contract. 

In an example like Angular, it's not as intuitive to override/set the `getScrollElement` method to satisfy existing logic that references that function in a variety of implementations.

i.e.:
```html
<cdk-virtual-scroll-viewport ion-content>
  <ion-refresher slot="fixed">
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher>
  ...
</cdk-virtual-scroll-viewport>
```

We aren't able to easily specify the scroll element, so the call inside `ion-refresher` to get the scroll element will throw an exception. 

Some current thoughts rolling around are:
- Expose a scroll element prop for each element coupled to `ion-content` functionality 
- Default scroll element to `getScrollElement()` or the `ion-content` host if the function isn't available
- Introduce an API method to set the scroll element and re-bind private `scrollEl` references

Just not currently excited that this is really copy/paste across implementations. Will explore other patterns/thoughts. We could potentially auto-detect the scroll target from the `ion-content` host, but that may not always be a good fit for certain implementations.

Reply:
Also we don't need to stress doing this as part of this PR 🙂  just more discussion details around it that the Ionic team will need to be aware of as it relates to this change.

Reply:
Yeah I totally feel like this PR is more of a "bring your own ion-content" implementation. For my use case, this would work as I'm ready to just create a new content component in my app and copy paste the necessary methods. 

I'm game for whatever you guys think would work best. Ultimately, I'm just trying to avoid shadow DOM and the issues that CSS variables cause for Safari. 🙂

Reply:
Ok, spent some time exploring an alternative pattern built on what you're doing in this PR, available here: https://github.com/ionic-team/ionic-framework/compare/FW-342

Explored with just a class name `.ion-content` (we will likely use a more specific name), but I believe we may decide to allow you to customize this selector from the `IonicConfig`, so that in your own implementation you could choose to target `[ion-content]` or whatever. 

Did a few optimizations to find the scroll target for you when not using `ion-content`. We're advancing this feature proposal to an internal review.

Solves two larger issues with one effort:
1. Webkit performance issue you are tracking.
2. Ability to use more of the Ionic eco-system in a virtual scroll context.

Reply:
Your implementation looks great 👍🏾. What you did in `content.utils.ts` is key!

Reply:
@sean-perkins just checking if there is any update?